src/util/taproot.rs

```diff
@@ -497,36 +499,38 @@ impl TaprootBuilder {
     }
 }
 
-// Internally used structure to represent the node information in taproot tree
+/// Data structure used to represent node information in taproot tree.
+/// You can use [`TaprootSpendInfo::from_node_info`] to a get [`TaprootSpendInfo`]
+/// from the merkle root [`NodeInfo`].
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub(crate) struct NodeInfo {
+pub struct NodeInfo {
     /// Merkle Hash for this node
     pub(crate) hash: sha256::Hash,
     /// information about leaves inside this node
     pub(crate) leaves: Vec<LeafInfo>,
 }
 
 impl NodeInfo {
-    // Create a new NodeInfo with omitted/hidden info
-    fn new_hidden(hash: sha256::Hash) -> Self {
+    /// Create a new [`NodeInfo`] with omitted/hidden info
+    pub fn new_hidden(hash: sha256::Hash) -> Self {
         Self {
             hash: hash,
             leaves: vec![],
         }
     }
 
-    // Create a new leaf with NodeInfo
-    fn new_leaf_with_ver(script: Script, ver: LeafVersion) -> Self {
+    /// Create a new leaf [`NodeInfo`] with given [`Script`] and [`LeafVersion`]
```

Comment:
nit
```suggestion
    /// Creates a new leaf [`NodeInfo`] with given [`Script`] and [`LeafVersion`]
```

Reply:
nit: nit: 
```suggestion
    /// Create a new leaf [`NodeInfo`] with given [`Script`] and [`LeafVersion`].
```
Phew, docs are hard to do meticulously :)